targets/XPlatCppSdk/source/code/source/playfab/PlayFabCurlHttpPlugin.cpp

```diff
@@ -64,10 +64,14 @@ namespace PlayFab
                     }
                 }
             }
-        }
-        catch (std::exception ex)
-        {
-            PlayFabPluginManager::GetInstance().HandleException(ex);
+            catch (const std::exception& ex)
```

Comment:
After looking at this change and [the previous PR](https://github.com/PlayFab/SDKGenerator/pull/419/files?utf8=%E2%9C%93&diff=unified&w=1) I'm not sure how you expect clients to  use these exception handling mechanisms. I think it would be really helpful for me to see some pseudo code or sample code demonstrating how you want clients will use it.

From my current understanding two problems are...
1. If I register to get background thread exceptions, how am I supposed to interpret what they mean?
```cpp
void ExceptionHandler(std::exception ex)
{
    // what can I do with ex here? How do I know what failed? was it an API call? was it some periodic background process that I can ignore?
   // if there's nothing I can meaningfully do with this exception isn't doesn't provide any value for me to catch it
}
```
If I call one of the SDKs APIs and have an exception thrown, that tells me that the API call failed. Depending on the error I can programmatically respond the exception by retrying at a later time or I can just toast a helpful error message to any end users.
```cpp
try
{
    PlayFab::PlayFabMultiplayerAPI::RequestMultiplayerServer(...);
}
catch(const std::exception& ex)
{
    // oh no! RequestMultiplayerServer failed! Better report it
    ReportFailure(FailureType::RequestMultiplayerServer, ex.what());
}
```

2. If some an exception is thrown, it could potentially lead to me as the caller never getting a completion callback. This means I have to write timeout logic for every playfab API call. This sounds really painful as dev :( I'd much prefer if these background exceptions were reported through the errorcallbacks that I already have to supply to the playfab SDK whenever I request some work be done. Any work that's not associated with an API call is probably fine to just log and ignore.